src/Datadog.Trace.ClrProfiler.Managed/Emit/MethodBuilder.cs

```diff
@@ -152,6 +158,15 @@ private TDelegate EmitDelegate()
                 string message = $"Unable to resolve method {_concreteTypeName}.{_methodName} by metadata token: {_mdToken}";
                 Log.Error(message, ex);
                 requiresBestEffortMatching = true;
+                _methodBase = null; // Be extra sure the assignment never happened
```

Comment:
The assignments in the `try` clause above will never happen if an exception is thrown calling `_resolutionAssembly.ManifestModule.ResolveMethod()`. Any chance `_methodBase` was set somewhere else?

Reply:
I didn't see anywhere, it's totally unlikely that it is the cause, but I'd prefer it to be extra clear and extra safe.